unlock-app/app/checkout/[id]/page.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import { Metadata } from 'next'
+import { fetchMetadata } from 'frames.js/next'
+import { CheckoutPage as CheckoutPageComponent } from '~/components/interface/checkout'
+import { getConfig } from '../../frames/checkout/components/utils'
+import { config as appConfig } from '~/config/app'
+import { notFound } from 'next/navigation'
+
+type Props = {
+  params: { id: string }
+}
+
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  const id = params?.id?.trim()
+
+  // Default metadata without frame data
+  const baseMetadata: Metadata = {
+    title: 'Checkout | Unlock Protocol',
+  }
+
+  // Return base metadata if no valid ID is present
+  if (!id || id.length === 0) {
+    return baseMetadata
+  }
+
+  try {
+    const config = await getConfig(id)
+
+    // Return base metadata if config could not be fetched
+    if (!config) {
+      return baseMetadata
+    }
+
+    const metadata: Metadata = {
+      title: config.title || baseMetadata.title,
+      openGraph: {
+        images: [`/og/checkout?id=${id}`],
+      },
+      other: {
+        ...(await fetchMetadata(
+          new URL(`/frames/checkout?id=${id}`, appConfig.unlockApp)
+        )),
+      },
+      alternates: {
+        canonical: `${appConfig.unlockApp}/checkout/${id}`,
+      },
+    }
+
+    return metadata
+  } catch (error) {
+    console.error('Error generating metadata:', error)
```

Comment:
Let's add the config Id here so we can more easily identify which is is not working.